### PR TITLE
[FLOSS H3] mozprefs: Fixing silent failing for mozprefs plugin

### LINKED ISSUE #4585: [FLOSS] mozprefs plugin fails silently on meta-set
As discussed in #4555
## Steps to Reproduce the Problem
```sh
$ sudo kdb mount prefs.js user:/tests/mozprefs mozprefs
# RET: 0
$ sudo kdb meta-set system:/tests/hosts somevalue a
# RET: 0  
$ sudo kdb meta-get system:/tests/hosts somevalue
Metakey not found
# RET: 2
```
## Expected Result

Should print an error on `sudo kdb mount prefs.js user:/tests/mozprefs mozprefs` and return a non 0 exit code

## System Information
Operating System:
Arch Linux, Kernel 5.19.13-arch1-1

kdb and plugins version
```sh
# kdb --version
KDB_VERSION: 0.9.11
SO_VERSION: 5
```

## Relevant Source Code
mozprefs ignores all metadata other than type

https://github.com/ElektraInitiative/libelektra/blob/cf4b6f303432dbfdcb9cc222c84f41045962b211/src/plugins/mozprefs/mozprefs.c#L210

### PR DESCRIPTION
**Pull request by Dynamichost96**
Fixes #4585

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [ ] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation (see [Documentation Guidelines](https://www.libelektra.org/devgettingstarted/documentation))
- [ ] I fixed all affected decisions (see [Decision Process](https://www.libelektra.org/decisions/decision-process))
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://www.libelektra.org/devgettingstarted/coding))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

**Review by janldeboer — COMMENTED** [review]
Good change!
Reading through the code all the changes seem appropriate to fix the mentioned issues.
Just run the format patches proposed by the `continuous-integration/jenkins/pr-merge` check to fix the format.
I wonder why only one check was run?

File: doc/news/_preparation_next_release.md [hunk 1]
```diff
@@ -187,10 +187,14 @@ The following text lists news about the [plugins](https://www.libelektra.org/plu
...
 ### csvstorage
 
-- Fix a bug where writing unkown meta keys will fail silently _(Juri Schreib @Bujuhu)_
+- Fix a bug where writing unknown meta keys will fail silently _(Juri Schreib @Bujuhu)_
 - <<TODO>>
 - <<TODO>>
 
+### mozprefs
+
+-Fixed bug when inserting Metakeys it would fail silently _(Nikola Prvulovic @Dynamichost96)_
```
Comment (flo91) [inline doc/news/_preparation_next_release.md:196]:
```suggestion
-Fixed bug: When inserting unknown meta keys it failed silently _(Nikola Prvulovic @Dynamichost96)_
```

File: src/plugins/mozprefs/mozprefs.c [hunk 2]
```diff
@@ -235,12 +235,36 @@ static void writeKey (FILE * fp, const Key * parentKey, const Key * key)
...
+	Key * cur = 0;
+	for (elektraCursor it = 0; it < ksGetSize (keySet); ++it)
+	{
+		cur = ksAtCursor (keySet, it);
+		const KeySet * metaKeys = keyMeta (cur);
+		for (elektraCursor jt = 0; jt < ksGetSize (metaKeys); ++jt)
+		{
+			const Key * meta = ksAtCursor (metaKeys, jt);
+			const char * pos = (const char *) keyName (meta);
+			if (elektraStrNCmp (pos, "meta:/internal/mozprefs", 19) != 0 && elektraStrCmp (pos, "meta:/origname") && elektraStrNCmp (pos, "meta:/rename", 12) != 0 && elektraStrCmp (pos, "meta:/binary") != 0)
```
Comment (flo91) [inline src/plugins/mozprefs/mozprefs.c:249]: Why did you only compare the first 19 characters in `elektraStrNCmp (pos, "meta:/internal/mozprefs", 19)`?

Please use a consistent coding style (`elektraStrCmp (pos, "meta:/origname")` was not explicitly compared against `0`).

```suggestion
			if (elektraStrNCmp (pos, "meta:/internal/mozprefs", 23) != 0 && elektraStrCmp (pos, "meta:/origname") != 0 && elektraStrNCmp (pos, "meta:/rename", 12) != 0 && elektraStrCmp (pos, "meta:/binary") != 0)
```

**Review by flo91 — CHANGES_REQUESTED** [review]
Thank you for the PR!
Please fix the styling issues as reported by the CI (you can apply the patch provided by the CI), clean up the commit history (rebase!) and answer the questions of my review.

**Pull request closed without merging**